rest-service/manager_rest/rest/resources_v3.py

```diff
@@ -387,3 +391,63 @@ def delete(self):
         """
         params = self._get_and_validate_params()
         return PermissionsHandler.remove_permissions(params)
+
+
+class LdapAuthentication(SecuredResource):
+    @rest_decorators.exceptions_handled
+    @rest_decorators.marshal_with(LdapResponse)
+    def post(self):
+        if not current_user.is_admin:
+            raise UnauthorizedError('User is not authorized to set LDAP '
+                                    'configuration.')
+        if not _is_manager_clean():
+            raise MethodNotAllowedError('LDAP Configuration may be set only on'
+                                        ' a clean manager.')
+        if not current_app.premium_enabled:
+            raise MethodNotAllowedError('LDAP is only supported in the '
+                                        'Cloudify premium edition.')
+
+        ldap_config = get_json_and_verify_params({'ldap_server',
+                                                  'ldap_username',
+                                                  'ldap_password',
+                                                  'ldap_domain',
+                                                  'ldap_is_active_directory',
+                                                  'ldap_dn_extra'})
+
+        from cloudify_premium.multi_tenancy.ldap_authentication \
+            import LdapAuthentication
+
+        # update current configuration
+        for key, value in ldap_config.iteritems():
+            setattr(rest_config.instance, key, value)
+
+        # assert LDAP configuration is valid.
+        authenticator = LdapAuthentication()
+        authenticator.configure_ldap()
+        try:
+            authenticator.authenticate_user(ldap_config.get('ldap_username'),
+                                            ldap_config.get('ldap_password'))
+        except UnauthorizedError:
+            # reload previous configuration.
+            rest_config.instance.load_configuration()
+            raise BadParametersError(
+                'Failed setting LDAP authenticator: Invalid parameters '
+                'provided.')
+
+        rest_config.reset(rest_config.instance, write=True)
+
+        # Restart the rest service so that each the LDAP configuration
+        # be loaded to all flask processes.
+        set_restart_task()
+
+        ldap_config.pop('ldap_password')
+        return ldap_config
+
+
+def _is_manager_clean():
```

Comment:
maybe _only_admin_in_manager or something like that, 
because manager might not be clean (might contain blueprint, deployments, etc.)